pkg/ha/ha_service.go

```diff
@@ -61,10 +63,15 @@ func (ha *HAService) setEndpoints(ctx context.Context) error {
 	endpoints := corev1.Endpoints{}
 	err := ha.manager.GetClient().Get(ctx, client.ObjectKey{Namespace: ha.namespace, Name: app.Name}, &endpoints)
 	if err != nil {
-		return fmt.Errorf("updating the service endpoint to point to the new leader: retrieving endpoints: %w", err)
+		if !errors.IsNotFound(err) {
+			return fmt.Errorf("updating the service endpoint to point to the new leader: retrieving endpoints: %w", err)
+		}
+
+		endpoints.ObjectMeta.Namespace = ha.namespace
+		endpoints.ObjectMeta.Name = app.Name
 	}
 
-	endpoints.ObjectMeta.Labels = map[string]string{"app": app.Name}
+	endpoints.ObjectMeta.Labels = map[string]string{"app": app.Name, "resources.gardener.cloud/managed-by": "gardener"}
```

Comment:
This label is added internally by GRM to the objects managed by a ManagedResource. See https://github.com/gardener/gardener/blob/c13d436b03a26bdf236a492db6c32649d9b62bbd/pkg/apis/resources/v1alpha1/types.go#L59-L61
Why we add this label here to the endpoint?

Reply:
It's not necessary, and I've already removed it on the gardenlet side. I intentionally postponed the change here, so I'd have a working code surface with all the latest changes, before I start introducing new changes.

Just removed in gardener-custom-metrics as well.